### PR TITLE
EZP-31028: Fixed formatting involving soft return placed inside a Custom Template

### PR DESCRIPTION
**Pull request by adamwojs**
| Question           | Answer
| ------------------ | ------------------
| **JIRA issue**     | [EZP-31028](https://jira.ez.no/browse/EZP-31028)
| **Bug/Improvement**| yes
| **New feature**    | no
| **Target version** | `1.1`
| **BC breaks**      | no
| **Tests pass**     | yes
| **Doc needed**     | no

### QA

- [ ] Sanity checks for nested content with soft line breaks inside Custom Tags.
- [ ] Sanity checks for nested content with soft line breaks inside Custom Styles.
- [ ] Double check RichText payload sent after publishing and editing again a content  (e.g. if single "&lt;br&gt;" is present).
- [ ] Regression test for [DEMO-305](https://jira.ez.no/browse/DEMO-305) ideally with re-publishing of chosen Content item.

**TODO**:
- [X] Implement feature / fix a bug.
- [X] Implement tests.
- [X] Fix new code according to Coding Standards (`$ composer fix-cs`).
- [X] Ask for Code Review.

### CODE REVIEW

File: tests/lib/eZ/RichText/Converter/Xslt/_fixtures/docbook/031-ezstyle.xml [hunk 1]
```diff
@@ -19,4 +19,11 @@ Highlighted block with multiple lines
 Highlighted block with multiple lines</literallayout>
         </ezcontent>
     </eztemplate>
+    <eztemplate name="highlighted_block" type="style">
+        <ezcontent>
+            <literallayout>Highlighted block with multiple lines and emphasis
+<emphasis role="strong">Highlighted block with multiple lines and emphasis</emphasis>
```
Comment (mikadamczyk) [inline tests/lib/eZ/RichText/Converter/Xslt/_fixtures/docbook/031-ezstyle.xml]: indentation?

File: tests/lib/eZ/RichText/Converter/Xslt/DebugRenderer.php [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+use EzSystems\EzPlatformRichText\eZ\RichText\RendererInterface;
+
+final class DebugRenderer implements RendererInterface
+{
+    private const TEMPLATE_FORMAT = '<template-output name="%s" type="%s" is-inline="%d">%s</template-output>';
+    private const EMBED_CONTENT_FORMAT = '<embed-content-output content-id="%d" view-type="%s" is-inline="%s">%s</embed-content-output>';
+    private const EMBED_LOCATION_FORMAT = '<embed-location-output location-id="%d" view-type="%s" is-inline="%s">%s</embed-location-output>';
+
+    public function renderTag($name, array $parameters, $isInline): string
```
[the remark this answers is not in the document]
Comment (adamwojs) [reply tests/lib/eZ/RichText/Converter/Xslt/DebugRenderer.php]: This method is part of the `RendererInterface` interface ;-)

[the remark this answers is not in the document]
Comment (mikadamczyk) [reply tests/lib/eZ/RichText/Converter/Xslt/DebugRenderer.php]: You are right :)

File: src/lib/eZ/RichText/Resources/stylesheets/docbook/xhtml5/output/core.xsl [hunk 3]
```diff
@@ -58,9 +58,10 @@
     </xsl:element>
   </xsl:template>
 
-  <xsl:template match="docbook:section/docbook:literallayout">
+
```
Comment (konradoboza) [inline src/lib/eZ/RichText/Resources/stylesheets/docbook/xhtml5/output/core.xsl]: Extra line not needed here.

**Comment by adamwojs** [comment]
PR updated according to @mikadamczyk and @konradoboza code review suggestions.

**Comment by adamwojs** [comment]
PR updated according to @alongosz code review suggestions.

**Comment by lserwatka** [comment]
Could you merge it up?

**Comment by adamwojs** [comment]
Done.

**Pull request merged**